text2sql-frontend/src/components/Providers/HealthcheckProvider.tsx

```diff
@@ -0,0 +1,65 @@
+import { createContext, useContext, useEffect, useState } from "react";
+import { api } from "../../api";
+import { enqueueSnackbar } from "notistack";
+
+type HealthCheckContextType = [boolean | null];
+
+const HealthCheckContext = createContext<HealthCheckContextType>([null]);
+
+// Custom hook that returns the user info and a function to set the avatar blob
+export const useHealthCheck = () => {
+  const context = useContext(HealthCheckContext);
+  if (context === undefined) {
+    throw new Error("useHealthCheck must be used within a HealthCheckProvider");
+  }
+  return context;
+};
+
+export const HealthCheckProvider = ({ children }: React.PropsWithChildren) => {
+  const [isHealthy, setIsHealthy] = useState<boolean | null>(null);
+
+  useEffect(() => {
+    const checkBackendHealth = async () => {
+      try {
+        console.log("Performing healthcheck");
+        await api.healthcheck();
+        if (isHealthy !== true) {
+          if (isHealthy === false) {
+            enqueueSnackbar({
+              variant: "success",
+              message: "Successfully reconnected to the backend.",
+            });
+          }
+          setIsHealthy(true);
+        }
+      } catch (e) {
+        if (isHealthy !== false) {
+          setIsHealthy(false);
+          enqueueSnackbar({
+            variant: "error",
+            message: "Failed to connect to the backend.",
+          });
+        }
+        return;
+      }
+    };
+
+    if (isHealthy === null) {
```

Comment:
This isHealthy will not be affected by the above setIsHealthy because this setting of state is async. So here you'll still be checking the old value which might not be expected.

I think even if this is intentional, it's pretty tricky code and we should simplify.

Reply:
It is intentional, and there's no setIsHealthy being called before this line